src/sys/windows/tcp.rs

```diff
@@ -397,6 +405,10 @@ impl StreamImp {
             self.inner.socket.write_overlapped(&buf[pos..], overlapped)
         };
         match err {
+            Ok(true) if me.instant_notify => {
+                me.write = State::Empty;
+                self.add_readiness(me, Ready::writable());
```

Comment:
@alexcrichton 
did some benchmarking to simply deterine the overhead of the `.result` call.
code: https://gist.github.com/steffengy/598350018d4827887b287ed0ea798850

mio/tcp.rs:
```patch
-self.inner.socket.result(overlapped).map(Some)
+Ok(Some((buf.len() - pos, 3u32)))
```


results (with .result call):

```
1,943,652 ns/iter (+/- 525,906)
2,009,193 ns/iter (+/- 1,153,769)
1,943,501 ns/iter (+/- 565,776)
```
without the .result call:
```
1,753,306 ns/iter (+/- 1,052,692)
1,724,071 ns/iter (+/- 589,609)
1,792,201 ns/iter (+/- 537,950)
```